client/signup/config/flows.js

```diff
@@ -188,7 +188,11 @@ function getDIFMSiteContentCollectionDestination( { siteSlug } ) {
 }
 
 function getSitesDestination( { siteSlug } ) {
-	return addQueryArgs( { 'new-site': siteSlug }, '/sites' );
+	if ( siteSlug ) {
+		return addQueryArgs( { 'new-site': siteSlug }, '/sites' );
+	}
+
+	return '/sites';
```

Comment:
Is this change necessary? I thought that if the value is undefined, the query parameter is omitted 🤔 

Oh, there are two `addQueryArg` functions in the codebase. One behaves as I described; the other does not.

Reply:
Quite right

https://github.com/Automattic/wp-calypso/blob/93a6d449e4dee3bbea8dc1fbbc0f3953927f8ea8/client/lib/url/add-query-args.ts#L28

It's not needed.

The reason I assumed it _wouldn't_ strip `null` or `undefined` values was because how else would you add a query param that was value-less. I guess that's just not possible with this `addQueryArgs` function 🤷